### ISSUE TITLE
Make DDP + Zero have the same communication volume as regular DDP

### ISSUE THREAD

**Issue by rohan-varma** [pytorch/pytorch#69591]
## 🚀 Feature

Currently when training DDP + ZeRo, we still reduce all gradients but ZeRo will update on each rank only the local parameter shard and then broadcast the rest. As a result, DDP only needs to reduce the gradients corresponding to that shard. In addition, zero redundancy optimizer does N broadcasts to share parameters across ranks: https://fburl.com/3o00xa81, but a single allgather could be done instead to improve performance.

**Comment by rohan-varma** [comment]
@mrshenli suggested 2 options:

1. In DDP's zero comm hook, use reduce_scatter instead of allreduce
2. Wait until Zero-1 support is added to FSDP, and then retire ZeroRedundancyOptimizer.

**Comment by pritamdamania87** [comment]
Wondering if it is worth doing 1. at this point instead of directly going for 2.?

**Comment by rohan-varma** [comment]
I believe @zhaojuanmao proposed adding Zero-1, Zero-2 support to FSDP as part of an effort to create a unified DDP/FSDP story. I think this can probably be absorbed into that timeline unless there is an urgent need for this, which I don't think there is at the moment.
